app/models/project.rb

```diff
@@ -39,4 +56,12 @@ def as_json(options = {})
   def available_members
     (User.all.select(:id, :name, :email) - users.select(:id, :name, :email))
   end
+
+  ##
+  # Get a list of projects that can be added as components to this project
+  def available_components
+    return [] unless Component.where(child_project_id: id).count.zero?
+
+    Project.where.not(id: [id] + component_projects.pluck(:child_project_id) + Component.pluck(:project_id))
+  end
```

Comment:
Is there anything stopping a user who does not have access to a project from adding it as a component to their project?

Reply:
@Bialogs I think previously said that anyone should be able to see projects, but need at least author to make any sort of changes to one. I'm not sure if that is still true so it would be worth checking on what the path forward is for project visibility to non-members.

Reply:
Thinking about this more, I feel like either the user who is exporting the project should have to have permission to see all of the components, or the user who is adding the project should have access to see all of the components, or maybe we just require both?

Reply:
probably should do both

Reply:
added filtering here, but the downside is that this method is pretty slow because of all the queries that it generates